app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.java

```diff
@@ -0,0 +1,154 @@
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import java.util.Collections;
+import java.util.List;
+
+import net.squanchy.R;
+import net.squanchy.schedule.domain.view.Event;
+import net.squanchy.schedule.domain.view.Schedule;
+import net.squanchy.schedule.domain.view.SchedulePage;
+import net.squanchy.schedule.view.EventItemView;
+import net.squanchy.schedule.view.EventViewHolder;
+import net.squanchy.schedule.view.ScheduleViewPagerAdapter;
+import net.squanchy.search.view.HeaderViewHolder;
+import net.squanchy.support.lang.Lists;
+
+import org.joda.time.LocalDateTime;
+
+class FavoritesAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private final LayoutInflater layoutInflater;
+
+    @IntDef(value = {ItemViewType.TYPE_TALK, ItemViewType.TYPE_HEADER})
+    @interface ItemViewType {
+
+        int TYPE_TALK = 0;
+        int TYPE_HEADER = 1;
+    }
+
+    private Schedule schedule = Schedule.create(Collections.emptyList());
+
+    @Nullable
+    private ScheduleViewPagerAdapter.OnEventClickedListener listener;
+
+    FavoritesAdapter(Context context) {
+        setHasStableIds(true);
+        layoutInflater = LayoutInflater.from(context);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return findFor(
+                0,
+                position,
+                schedule.pages(),
+                schedulePage -> (long) -schedulePage.dayId().hashCode(),
+                (schedulePage, positionInPage) -> schedulePage.events().get(positionInPage).numericId()
+        );
+    }
+
+    void updateWith(Schedule schedule, ScheduleViewPagerAdapter.OnEventClickedListener listener) {
+        this.schedule = schedule;
+        this.listener = listener;
+        notifyDataSetChanged();
+    }
+
+    @Override
+    @ItemViewType
+    public int getItemViewType(int position) {
+        return findFor(
+                0,
+                position,
+                schedule.pages(),
+                schedulePage -> ItemViewType.TYPE_HEADER,
+                (schedulePage, positionInPage) -> ItemViewType.TYPE_TALK
+        );
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, @ItemViewType int viewType) {
+        if (viewType == ItemViewType.TYPE_TALK) {
+            EventItemView itemView = (EventItemView) layoutInflater.inflate(R.layout.item_schedule_event_talk, parent, false);
+            return new EventViewHolder(itemView);
+        } else if (viewType == ItemViewType.TYPE_HEADER) {
+            return new HeaderViewHolder(layoutInflater.inflate(R.layout.item_search_header, parent, false));
+        } else {
+            throw new IllegalArgumentException("View type not supported: " + viewType);
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        if (holder instanceof EventViewHolder) {
+            Event event = findFor(
```

Comment:
Same here, if we have the data already in the format we need, we don't have to find it every time